### DISCUSSION TITLE
How to fix "No modules to process in combine-js-to-schema-cli."

### DISCUSSION THREAD

**Discussion by silencer07** [reactwg/react-native-new-architecture#210] [Q&A]
Hi everyone!

I am trying to make a sample Turbo Module. This is basically a continuation of  https://github.com/reactwg/react-native-new-architecture/discussions/200 but with a different error

I am encountering error:

> Task :rtn-splash-screen:generateCodegenSchemaFromJavaScript
> No modules to process in combine-js-to-schema-cli. If this is unexpected, please check if you set up your NativeComponent correctly. See combine-js-to-schema.js for how codegen finds modules.

Weirdly enough the generate code command succeeds but no actual generated code is produced. I think this is a false positive.

link to the github action build: https://github.com/silencer07/codgen-error-reproducer-react-native/actions/runs/10258030324/job/28380178701

link to the reproducer app: https://github.com/silencer07/codgen-error-reproducer-react-native

Thanks to @cortinico for helping me with the previous error!

**Comment by cortinico** [comment]
It seems like your app is failing to compile instead:

```
> Task :rtn-splash-screen:compileDebugKotlin FAILED
e: file:///home/runner/work/codgen-error-reproducer-react-native/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/SplashScreenPackage.kt:9:3 'getModule' overrides nothing
```

**Reply by silencer07** [reply]
Hi @cortinico ,

I cannot do the override yet because the code generation did not work :( .

i.e. No SplashScreenModule file has been produced.

That's why i find it odd that codegen succeeded without being able to produce the code

edit: look at the line 43 of the github action log. that precedes the kotlin compilation step

**Comment by silencer07** [comment]
Update: I tried something like making the package name all small to adhere java standards, the error in `line 43` went away but still no code

**Reply by cortinico** [reply]
@silencer07 have you unblocked yourself or do you still need further support?

**Reply by silencer07** [reply]
@cortinico unfortunately no. We parked the migration for now. We will be trying again next year.

I will post my updates later on when I tried using it against newer version of react native

**Reply by silencer07** [reply]
OK what I did is created react native from start and followed the wiki to the letter. I experienced different error. Will close this discussion

**Comment by bumjin1013** [comment]
same issue

**Comment by nickplee** [comment]
@bumjin1013 @silencer07 do your modules names' start with the word `Native` or end with `NativeComponent`? If not, this check will fail and no code will be generated. 2+ hours I will never get back.
<img width="626" alt="image" src="https://github.com/user-attachments/assets/c99b3232-7cf2-4e5d-94ab-6cb2d4500678" />

**Reply by silencer07** [reply]
@nickplee 

Hi, please refer to working example of a turbo module here: https://github.com/silencer07/react-native-new-architecture

better yet use this instead for better experience: https://nitro.margelo.com/

**Reply by souzasmatheus** [reply]
@nickplee I spent about 4 hours, but you put an end to it. Thank you very much!

**Reply by SericCocom** [reply]
I'm having the same problem, but I didn't understand what the solution is, I'm new to this.
